osaca/parser/parser_x86att.py

```diff
@@ -308,84 +293,123 @@ def process_operand(self, operand):
+                name=index["name"], prefix_id=index["prefix"] if "prefix" in index else None
+            )
+        if isinstance(offset, dict) and "identifier" in offset:
+            offset = IdentifierOperand(name=offset["identifier"]["name"])
+        new_dict = MemoryOperand(
+            offset_ID=offset, base_id=baseOp, index_id=indexOp, scale_id=scale
+        )
         # Add segmentation extension if existing
-        if self.SEGMENT_EXT_ID in memory_address:
-            new_dict[self.SEGMENT_EXT_ID] = memory_address[self.SEGMENT_EXT_ID]
-        return AttrDict({self.MEMORY_ID: new_dict})
+        if self.segment_ext_id in memory_address:
+            new_dict.segment_ext_id = memory_address[self.segment_ext_id]
+        return new_dict
 
     def process_label(self, label):
         """Post-process label asm line"""
         # remove duplicated 'name' level due to identifier
         label["name"] = label["name"][0]["name"]
-        return AttrDict({self.LABEL_ID: label})
+        return LabelOperand(
+            name=label["name"], comment_id=label["comment"] if "comment" in label else None
+        )
 
     def process_immediate(self, immediate):
         """Post-process immediate operand"""
         if "identifier" in immediate:
             # actually an identifier, change declaration
-            return immediate
+            return self.process_identifier(immediate["identifier"])
         # otherwise just make sure the immediate is a decimal
-        immediate["value"] = int(immediate["value"], 0)
-        return AttrDict({self.IMMEDIATE_ID: immediate})
+        # immediate["value"] = int(immediate["value"], 0)
+        new_immediate = ImmediateOperand(value_id=int(immediate["value"], 0))
+        return new_immediate
+
+    def process_identifier(self, identifier):
+        return IdentifierOperand(name=identifier["name"])
 
     def get_full_reg_name(self, register):
         """Return one register name string including all attributes"""
         # nothing to do
-        return register["name"]
+        return register.name
 
     def normalize_imd(self, imd):
         """Normalize immediate to decimal based representation"""
-        if "value" in imd:
-            if isinstance(imd["value"], str):
+        if isinstance(imd, IdentifierOperand):
+            return imd
+        if imd.value is not None:
+            if isinstance(imd.value, str):
                 # return decimal
-                return int(imd["value"], 0)
+                return int(imd.value, 0)
             else:
-                return imd["value"]
+                return imd.value
         # identifier
         return imd
 
     def is_flag_dependend_of(self, flag_a, flag_b):
         """Check if ``flag_a`` is dependent on ``flag_b``"""
         # we assume flags are independent of each other, e.g., CF can be read while ZF gets written
         # TODO validate this assumption
+        if isinstance(flag_b, Operand):
```

Comment:
If we can't be sure anymore that `flag_b` is an `Operand`, can we be sure for `flag_a` or should this be checked as well? Also, what happens if `"name"` is not in `flag_b`, there should be a `if "name" in flag_b` check that returns False otherwise.
Furthermore, the checks in `parser_AArch64.py` are different to this one. I think both functions need a proper re-write.